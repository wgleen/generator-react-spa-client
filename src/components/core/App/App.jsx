import React from 'react'
import * as reducks from '../../../reducks'
import ReduxProvider from '../Redux/ReduxProvider'
import ThemeProvider from '../Theme/ThemeProvider'
import Router from '../Router/Router'

const App = () => (
  <>
    <ReduxProvider store={reducks.store}>
      <ThemeProvider>
        <Router />
      </ThemeProvider>
    </ReduxProvider>
  </>
)

export default App
