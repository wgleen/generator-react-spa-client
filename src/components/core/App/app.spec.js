import React from 'react'
import { render, screen } from '@testing-library/react'
import App from './App'

describe('<App />', () => {
  it('should render correctly', () => {
    expect(render(<App />)).toMatchSnapshot()
  })

  it('should render a text equal to "Hello World"', () => {
    render(<App />)

    expect(screen.getByText('Hello World')).toBeDefined()
  })
})
