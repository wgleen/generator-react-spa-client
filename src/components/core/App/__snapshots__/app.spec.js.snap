// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
<App>
  <ReduxProvider>
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <Router>
        <RouterProvider>
          <BrowserRouter>
            <Router
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
            >
              <Routes>
                <Suspense
                  fallback="Loading"
                >
                  <Switch>
                    <Route
                      component={
                        Object {
                          "$$typeof": Symbol(react.lazy),
                          "_init": [Function],
                          "_payload": Object {
                            "_result": Promise {},
                            "_status": 0,
                          },
                        }
                      }
                      computedMatch={
                        Object {
                          "isExact": true,
                          "params": Object {},
                          "path": "/",
                          "url": "/",
                        }
                      }
                      exact={true}
                      location={
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        }
                      }
                      path="/"
                    />
                  </Switch>
                  Loading
                </Suspense>
              </Routes>
            </Router>
          </BrowserRouter>
        </RouterProvider>
      </Router>
    </Provider>
  </ReduxProvider>
</App>
`;
